lib/Interpreter/CIFactory.cpp

```diff
@@ -297,8 +318,18 @@ namespace {
 
     #if defined(__GLIBCXX__)
       // Avoid '__float128 is not supported on this target' errors
-      if (!opts.Language && !opts.StdVersion)
-        sArguments.addArgument("-std=c++11");
+      if (!opts.Language && !opts.StdVersion) {
+        switch (CxxStdCompiledWith()) {
+          case 17:
+            // Hopefully -D__float128=void can be removed when -std=c++17 works.
+            sArguments.addArgument("-D__float128=void");
```

Comment:
Yes it's when cling is complied on OS X with gcc-6 with -std=c++1z.
I need this configuration (gcc on OS X) to work and would like to start testing it against c++17 and this issue showed up again.  I tried to at least move the c++17 test to  gcc-7 but there were even more errors.

I don't know of any better solution, but I'm pretty sure it's not a common configuration.
If it really bothers you just cut it out. But I think Travis is testing this.